ldap3/extend/microsoft/persistentSearch.py

```
@@ -0,0 +1,125 @@
+#
+# You should have received a copy of the GNU Lesser General Public License
+# along with ldap3 in the COPYING and COPYING.LESSER files.
+# If not, see <http://www.gnu.org/licenses/>.
+
+try:
+    from queue import Empty
+except ImportError:  # Python 2
+    # noinspection PyUnresolvedReferences
+    from Queue import Empty
+
+from ...core.exceptions import LDAPExtensionError
+from ...utils.dn import safe_dn
+from ...protocol.microsoft import persistent_search_control
+
+
+class ADPersistentSearch(object):
+    def __init__(self,
+                 connection,
+                 search_base,
+                 search_filter,
+                 search_scope,
+                 dereference_aliases,
+                 attributes,
+                 size_limit,
+                 time_limit,
+                 streaming,
+                 callback
+                 ):
+        if connection.strategy.sync:
+            raise LDAPExtensionError('Persistent Search needs an asynchronous streaming connection')
+
+        if connection.check_names and search_base:
+            search_base = safe_dn(search_base)
+
+        self.connection = connection
+        self.message_id = None
+        self.base = search_base
+        self.filter = search_filter
+        self.scope = search_scope
+        self.dereference_aliases = dereference_aliases
+        self.attributes = attributes
+        self.size_limit = size_limit
+        self.time_limit = time_limit
+        self.connection.strategy.streaming = streaming
+        if callback and callable(callback):
+            self.connection.strategy.callback = callback
+        elif callback:
+            raise LDAPExtensionError('callback is not callable')
+
+        self.controls = [persistent_search_control()]
+
+        self.start()
+
+    def start(self):
+        if self.message_id:  # persistent search already started
+            return
+
+        if not self.connection.bound:
+            self.connection.bind()
```

does it make sense to implicitly bind here if the connection isn't using a lazy binding strategy? it also doesn't look like we're checking the result of the bind
should this operation log something if it needs to handle binding? or should it fail for an unbound connection? 

regardless of whether this operation implicitly binds or not, it seems like if we get past this point and the connection is not bound (e.g. we fail to bind here), we should raise an exception rather than continue onwards to take the lock and search

---

You may be right here. This is just a copy-paste from the [standard peristent search method](https://github.com/cannatag/ldap3/blob/4d33166f0869b929f59c6e6825a1b9505eb99967/ldap3/extend/standard/PersistentSearch.py#L91). Whatever is done, the behaviour should be the same on both methods.

---

keeping the behavior consistent makes sense to me